File: jungclaire/leetcode/linked-list/141. Linked List Cycle.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 * int val;
 * ListNode next;
 * ListNode(int x) {
 * val = x;
 * next = null;
 * }
 * }
 */
public class Solution {
    public boolean hasCycle(ListNode head) {

        ListNode runner_1 = head;
        ListNode runner_2 = head;

        if (head == null) {
            return false;
        }
        if (head.next == null) {
            return false;
        }

        while (runner_2.next != null && runner_2.next.next != null) {
            runner_1 = runner_1.next;
            runner_2 = runner_2.next.next;

            if (runner_1 == runner_2) {
                return true;
            }
        }
        return false;
    }
}

File: 100milliongold/leetcode/linked-list/141.Linked List Cycle.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} head
 * @return {boolean}
 이게 되네???
 */
var hasCycle = function(head) {
    var list= []
    if(head === null )
        return false
    while( head = head.next ){
        if( list.indexOf(head) !== -1){
            return true
        }
        list.push(head);            
    }
    return false
};